Data/Aeson.hs

```diff
@@ -220,6 +224,15 @@ eitherDecodeStrict' =
   eitherFormatError . eitherDecodeStrictWith jsonEOF' ifromJSON
 {-# INLINE eitherDecodeStrict' #-}
 
+-- | Decode a nested JSON-encoded string.
+embeddedJSON :: (FromJSON a) => String -> (Value -> Parser a)
```

Comment:
Accepting a `Value` here is a bit strange since it will only suceed for strings, right? Is there a reason for it? A usage example would be helpful.

Reply:
It should take an aeson `String` instead?

Reply:
There is no `String` type in aeson, so it would take a `Text` instead.

Reply:
Right, that's what I meant. Sorry, I'm running a bit sleep-deprived at the moment.